frontend/src/components/ui/PasswordInput/PasswordInput.test.tsx

```
@@ -0,0 +1,413 @@
+    });
+
+    it('toggle button does not interfere with form submission', () => {
+      render(<PasswordInput />);
+      const toggleButton = screen.getByRole('button');
+      expect(toggleButton).toHaveAttribute('type', 'button');
+    });
+
+    it('input is keyboard accessible', async () => {
+      render(
+        <div>
+          <button>Before</button>
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)